### ISSUE TITLE
LT09 false positive - TOP, DISTINCT clause

### ISSUE THREAD

**Issue by bujocek** [sqlfluff/sqlfluff#7562]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The LT09 rule is treating TOP(x) or DISTINCT clause after SELECT keyword also as a select target. Which result in false positive identification as LT09 violation.

### Expected Behaviour

Nontarget keywords  should be ignored by the rule.

### Observed Behaviour

<img width="1029" height="340" alt="Image" src="https://github.com/user-attachments/assets/eb084773-6388-477e-889f-3ddb643e75cb" />

### How to reproduce

```
SELECT DISTINCT
  Id
FROM
  dbo.Locations

SELECT TOP (1)
  Id
FROM
  dbo.Locations
```

### Dialect

T-SQL

### Version

4.0.4

### Configuration

[sqlfluff]
dialect = tsql
exclude_rules = LT05, CP05, CP02, AL03, LT02, AM05, L022, ST06
warnings = LT01, LT09, RF05, LT12, AM04
large_file_skip_byte_limit = 35000

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 2
indented_joins = True

[sqlfluff:rules:structure.join_condition_order]
preferred_first_table_in_join_clause = later

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
`LT09` makes sure that select targets are always in a new line, UNLESS there is ONLY a single select target, in that case it allows to be a single line. In you examples there is only a single select target `id` so that should NOT be on a new line according to `LT09`, and that's the reason you get the linting error - which is correct IMHO.

If you fix this:

```sql
SELECT DISTINCT
  Id
FROM
  dbo.Locations
```

after the fix it will be on a single line:
```sql
SELECT DISTINCT Id
FROM
    Dbo.Locations
```

if you have more than a single select target it will handle that as well properly:

```sql
SELECT DISTINCT Id, col1
FROM
    Dbo.Locations
``` 

after the fix it becomes correclty:
```sql
SELECT DISTINCT
    Id,
    Col1
FROM
    Dbo.Locations
```

So I don't see here any false positive.

**Comment by bujocek** [comment]
Yes, you are right. My bad - I didn't check it correctly. Sorry for that.
